### PR TITLE
`azurerm_storage_account` - fix crash in multichannel checking

### LINKED ISSUE #19220: SIGSEGV with azurerm_storage_account creation when empty share_properties block is provided.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.4

### AzureRM Provider Version

3.30.0

### Affected Resource(s)/Data Source(s)

azurerm_storage_account

### Terraform Configuration Files

```hcl
terraform {
  required_version = ">= 1.3.0"

  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~> 3.0"
    }
  }
}

provider "azurerm" {
  features {}
}

resource "azurerm_storage_account" "myaccount" {
  resource_group_name      = var.rgName
  name                     = var.name
  location                 = "westus2"
  account_tier             = "Standard"
  account_replication_type = "LRS"
  account_kind             = "StorageV2"

  share_properties {
  }
}

variable "rgName" {
  default = "awang"
} 

variable "name" {
  default = "awangtestshare"
}
```

### Debug Output/Panic Output

```shell
azurerm_storage_account.myaccount: Creating...
azurerm_storage_account.myaccount: Still creating... [10s elapsed]
azurerm_storage_account.myaccount: Still creating... [20s elapsed]
azurerm_storage_account.myaccount: Still creating... [30s elapsed]
╷
│ Error: Request cancelled
│ 
│   with azurerm_storage_account.myaccount,
│   on main.tf line 16, in resource "azurerm_storage_account" "myaccount":
│   16: resource "azurerm_storage_account" "myaccount" {
│ 
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-azurerm_v3.30.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x106b7428c]

goroutine 136 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/storage.resourceStorageAccountCreate(0x140015b8f78?, {0x107944ca0?, 0x14000754800})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/storage/storage_account_resource.go:1317 +0x306c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x108381240?, {0x108381240?, 0x14000be1b90?}, 0xd?, {0x107944ca0?, 0x14000754800?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:695 +0x138
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14001679960, {0x108381240, 0x14000be1b90}, 0x1400194b790, 0x14000ebdd00, {0x107944ca0, 0x14000754800})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:837 +0x874
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x14000788900, {0x108381240?, 0x14000be1ad0?}, 0x1400224b2c0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/grpc_provider.go:1021 +0xb94
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x14000ab7860, {0x108381240?, 0x14000be1380?}, 0x14001948150)
	github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/tf5server/server.go:813 +0x370
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x107f8ef20?, 0x14000ab7860}, {0x108381240, 0x14000be1380}, 0x140015ae960, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x174
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000138a80, {0x1083934d8, 0x14000bd4680}, 0x140019386c0, 0x140017d8120, 0x10c4e7400, 0x0)
	google.golang.org/grpc@v1.47.0/server.go:1283 +0xb64
google.golang.org/grpc.(*Server).handleStream(0x14000138a80, {0x1083934d8, 0x14000bd4680}, 0x140019386c0, 0x0)
	google.golang.org/grpc@v1.47.0/server.go:1620 +0x840
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.47.0/server.go:922 +0x88
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.47.0/server.go:920 +0x298

Error: The terraform-provider-azurerm_v3.30.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behaviour

shoudl work or error out gracefully

### Actual Behaviour

SIGSEGV

### Steps to Reproduce

use terraform apply against the provided main.tf.

The culprit is this block
```
  share_properties {
  }
```

If i remove it it's fine.  That's a reasonable solution but I wanted to report  this bug as a SIGSEGV crash is not ideal error handling.

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by jackofallops**
Fixes #19220

### CODE REVIEW

File: internal/services/storage/storage_account_resource_test.go [hunk 1]
```diff
@@ -4107,3 +4122,28 @@ resource "azurerm_storage_account" "test" {
...
+resource "azurerm_storage_account" "test" {
+  name                = "unlikely23exst2acct%[3]s"
+  resource_group_name = azurerm_resource_group.test.name
+
+  location                 = azurerm_resource_group.test.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+  account_kind             = "StorageV2"
+
+  share_properties {}
```
Comment (tombuildsstuff) [inline internal/services/storage/storage_account_resource_test.go:4146]: the `share_properties` block also wants to become un-computed fwiw?

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
Left a few comments inline but otherwise 👍

File: internal/services/storage/storage_account_resource.go [hunk 2]
```diff
@@ -1336,17 +1336,22 @@ func resourceStorageAccountCreate(d *pluginsdk.ResourceData, meta interface{}) e
...
-				if *shareProperties.FileServicePropertiesProperties.ProtocolSettings.Smb.Multichannel.Enabled {
-					return fmt.Errorf("`multichannel_enabled` isn't supported for Standard tier Storage accounts")
+			smb := shareProperties.FileServicePropertiesProperties.ProtocolSettings.Smb
+			if smb != nil && smb.Multichannel != nil {
+
+				if smb.Multichannel.Enabled != nil {
+					if *shareProperties.FileServicePropertiesProperties.ProtocolSettings.Smb.Multichannel.Enabled {
+						return fmt.Errorf("`multichannel_enabled` isn't supported for Standard tier Storage accounts")
+					}
 				}
```
[the remark this answers is not in the document]
Comment (jackofallops) [reply internal/services/storage/storage_account_resource.go:1351]: This is dual purpose here, we need to error on `*smb.Multichannel.Enabled == true` and zero the `*smb.Multichannel` regardless, hence the split of `if` conditions, so I think this is correct as is?

File: internal/services/storage/storage_account_resource.go [hunk 3]
```diff
@@ -1336,17 +1336,22 @@ func resourceStorageAccountCreate(d *pluginsdk.ResourceData, meta interface{}) e
 		fileServiceClient := meta.(*clients.Client).Storage.FileServicesClient
 
 		shareProperties := expandShareProperties(val.([]interface{}))
+
 		// The API complains if any multichannel info is sent on non premium fileshares. Even if multichannel is set to false
-		if accountTier != string(storage.SkuTierPremium) {
+		if accountTier != string(storage.SkuTierPremium) && shareProperties.FileServicePropertiesProperties.ProtocolSettings != nil {
```
[the remark this answers is not in the document]
Comment (jackofallops) [reply internal/services/storage/storage_account_resource.go:1341]: Good shout

File: internal/services/storage/storage_account_resource.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply internal/services/storage/storage_account_resource.go:1352]: sorry missed this notification - sounds good/works for me

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Pull request merged**